### ISSUE TITLE
ci: Nix enhancements

### ISSUE THREAD

**Issue by realeinherjar** [bitcoindevkit/bdk_wallet#126]
Some enhancements were suggested in bitcoindevkit/bdk#1165.
They are non-blockers and I am moving them to a separate issue for the near future.

- [ ] Move to [`rustshop/flakebox`](https://github.com/rustshop/flakebox)
   - It can build derivations in parallel with `pkgs.linkFarmFromDrvs` ([example](https://github.com/rustshop/flakebox/blob/5dea81cad2be257a2cbccbd42b2f1c0fdf69008d/checks/workspace-cross-compile/default.nix#L93), from https://github.com/bitcoindevkit/bdk/pull/1165#discussion_r1374071091)
   - It will also use `mold` by default as the Linux linker. It can speed up builds.

- [ ] Upload failed derivations as artifacts.
   From [`fedimint/fedimint`](https://github.com/fedimint/fedimint/blob/0ef5039086a4cd863484cbfacd638d8a0d42c233/.github/workflows/ci-nix.yml#L126) in https://github.com/bitcoindevkit/bdk/pull/1165#discussion_r1374078143
   ```yaml
   - name: Prepare failed test build dirs
     if: always()
     run:  |
       set -x
       # delete source and target artifacts as it is huge and rather useless now
       sudo rm -Rf /tmp/nix-build-*/source || true
       # chown so actions/upload-artifact can access it
       sudo chown -R $USER /tmp/nix-build-* || true
       # delete unix sockets, as actions/upload-artifact can't handle them
       find /tmp/nix-build-* -type s -print0 | xargs -0 rm || true

    - name: Upload failed test build dirs
      if: always()
      uses: actions/upload-artifact@v3
      with:
        name: "fedimint-failed-test-logs-${{ github.run_number }}-${{ matrix.host}}"
        path: |
          /tmp/nix-build-*/

- [ ] Auto-correct typos.
   This also involves having a `just` setup.
   Example from https://github.com/fedimint/fedimint/blob/0ef5039086a4cd863484cbfacd638d8a0d42c233/justfile#L80, in https://github.com/bitcoindevkit/bdk/pull/1165#issuecomment-1782308478

- [ ] Use `cargo-nextest`. https://github.com/bitcoindevkit/bdk/pull/1165#issuecomment-1782278992

**Comment by luisschwab** [comment]
Closing this as not planned (see [bdk:#1320](https://github.com/bitcoindevkit/bdk/pull/1320) for context).
